### DISCUSSION TITLE
Make smart induction cooker

### DISCUSSION THREAD

**Discussion by rj10rd** [arendst/Tasmota#13587] [Q&A] [answered]
I have induction cooker and it's motherboard have highway09a ic.its control panel communicate with ic by i2c protocol.please tell me how use esp32 and tasmota to control it.

**Comment by ascillato** [comment, marked as answer]
Hi, that hardware is not supported in Tasmota.

If you want to control it, you will have to design everything from scratch. Sorry.

**Reply by rj10rd** [reply]
Sorry for what
Tasmota is awesome 😎.

**Reply by sfromis** [reply]
With ESP32, Tasmota has the Berry language which is nice for controlling stuff, including the ability to code I2C drivers.

**Reply by rj10rd** [reply]
How debug i2c protocol

**Comment by barbudor** [comment]
This discussion may help you understand how the  cooker works in general and what is the role of the 'highway' chip.

If you want to reverse engineer I2C prococol (in case the documentation provided in the link is not enough) you will need a logic analyser (basic ones can be purchased on AliExpress for less than 10€/$ are enough) + software for analysing
Search on Google for that

**Reply by rj10rd** [reply]
Can i use tasmota as a momentary button

**Reply by barbudor** [reply]
if you want a relay to generate a short pulse you can configure PulseTime
Then the "POWER ON" command will turn the relay ON for this short time and turn OFF immediately after

**Reply by rj10rd** [reply]
What i want:-
I open my induction cooker and saw there are two ic 
One is on main motherboard and other is on control panel board and they are communicate with i2c protocol. Momentary button attached with control panel board.
So i want replace momentary button with tasmota. For controlling cooker.

**Reply by sfromis** [reply]
If the momentary button is a simple mechanical button, and the control panel board has common ground with the ESP board, configuring a pin as Relay or Relay_i should indeed provide the ability of simulating the activation of such a button. If it is a touch button, things may be more complex.

**Reply by rj10rd** [reply]
Please tell me if anything available like a mini/micro relay

**Reply by sfromis** [reply]
Do you mean for simulating the button? Configuring the *pin* as a Relay does not mean that you need a physical relay.

**Reply by rj10rd** [reply]
Please elaborate. I m very beginner.
